DRAFT_homogeneous_impact_areas.Rmd

```diff
@@ -0,0 +1,426 @@
+infra_trd_ll <- sf::st_transform(infra_trd, 4326)
+```
+
+Get the boundary box of the cropped raster
+
+```{r, eval=F}
+myBB_ll <- sf::st_bbox(infra_trd_ll)
+```
+
+Download highways using the above bbox.
+
+```{r, warning=FALSE, message=F, eval=F}
+hw <- 
+  osmplotr::extract_osm_objects(
+    bbox = myBB_ll,
+    key = "highway",
+    sf = T)
+```
+
+Transforming highway data back into utm, although not strictly necessary.
+
+```{r, eval=F}
+hw_utm <- sf::st_transform(hw, sf::st_crs(infra_trd)) 
+```
+
+This object contains too many roads (about 30k).
+I'll take out the unnamed roads.
+
+```{r, eval=F}
+hw_utm <- hw_utm[!is.na(hw_utm$name),]
+```
+
+```{r, eval=F, echo=F}
+# Caching the sf object
+saveRDS(hw_utm, "data/cache/highways_trondheim.rds")
+```
+
+```{r, echo=F}
+#Load object
+hw_utm <- readRDS("data/cache/highways_trondheim.rds")
+```
+
+#### Discretize
+
+Here I define a simplified categorical typology for the infrastructure index using four classes.
+
+```{r}
+names(infra_trd) <- "infrastructureIndex"
+infra_trd_reclassed <-  infra_trd %>%
+  mutate(infrastructureIndex = case_when(
+    infrastructureIndex < 1 ~ 0,
+    infrastructureIndex < 6 ~ 1,
+    infrastructureIndex < 12 ~ 2,
+    infrastructureIndex >= 12 ~ 3
+  ))
+```
+
+Lets plot these two maps side by side
+
+```{r}
+map_trd_reclassed <- tm_shape(infra_trd_reclassed)+
+  tm_raster(title="Infrastructure Index",
+            #palette = "viridis",
+            style="cat")+
+  tm_layout(legend.outside = T)+
+  tm_shape(hw_utm)+
+  tm_lines(col="white")
+```
+
+```{r}
+map_trd <- tm_shape(infra_trd)+
+  tm_raster(title="Infrastructure Index",
+            style="cont",
+            palette = "viridis")+
+  tm_layout(legend.outside = T)+
+  tm_shape(hw_utm)+
+  tm_lines(col="white")
+```
+
+```{r, fig.cap="Infrastructure indeks over Trondheim, comparing the continous scale with the ordinal four-step scale. Major roads are in white."}
+tmap_arrange(map_trd,
+             map_trd_reclassed,
+             ncol=1)
+```
+
+I tweaked the thresholds for the bins so that the categories match my knowledge about the land use intensity in Trondheim, for example that most of the forest next to Trondheim (to the left in the map) was in the second lowest class.
+This looks pretty good to me.
+It depicts a gradient in human presence from high within the buildt zone, to *no to very low* in the forests and mountains to the west.
+Note that there is still considerable human activity also there in the form of outdoor recreation and even forestry.
+
+#### Aggregate
+
+The resolution in this map is more than we need, and the size of the data implies that the next step, the vectorization, would take too long.
+We therefore aggregate, ar reduce the resolution.
+
+```{r}
+dim <- st_dimensions(infra)
+paste("Resolution is", dim$x$delta, "by", dim$x$delta, "meters")
+```
+
+```{r, warning=F, message=F}
+infra_trd_reclassed_agg <- st_warp(infra_trd_reclassed, cellsize = c(1000, 1000), 
```

Comment:
I get an error here: Error in inherits(dest, "stars_proxy") : 
  argument "dest" is missing, with no default

Reply:
It runs locally on my machine with stars v. 0.6-0

Reply:
Fixed by updating the package